### ISSUE TITLE
improve CI output in summary: More informative

### ISSUE THREAD

**Issue by tbates** [OpenMx/OpenMx#78]
from #75 

1. Include values when the estimate produces one (don't convert to NA just because of warnings)
2. enhance the output column, so instead of just  `!!!`  add a brief code for the error (perhaps as simple as 1, 2, 3... 
3. list these coded outcomes in an attribute, which the print method for CIs can display, but doesn't bollox the table for users seeking to post-process it.

candidate codes could be verbal, so 
`!!! [alpha, OK]]` for "problem: lower bound alpha level not reached"
`!!! [OK, constrained]]` for "problem: "upper had non-linear constraint infeasible"
`!!! [blue, blue]]` for "problem: "lower iteration limit/blue, upper iteration limit/blue" 
`!!! [blue, ]]` for "problem: "lower iteration limit/blue, upper not requested" 

This would save users from having to mxSummary(model, verbose=T) to see what's up.

**Comment by RMKirkpatrick** [comment]
The existing 'CI details' table would be improved simply by making the first seven columns the "parameter", "side", "value", "fit", "diagnostic", "statusCode", and "method" columns, in that order.

That's an easy change to make.  Does anyone object?

**Comment by jpritikin** [comment]
Yeah, the difficulty is caused by the desire to combine the upper and lower bounds on a single line.

**Comment by tbates** [comment]
Details is probably fine: this issue #78 is about formatting summary compactly

**Comment by jpritikin** [comment]
Hm, did we decide how to improve the output? Reading the above exchange, there are some ideas but it is not clear whether we reached a consensus.

**Comment by RMKirkpatrick** [comment]
Having thought about it, I think Tim's suggestion is mostly sensible.  I was initially resistant to placing the diagnostic info for both confidence limits on one line, but I think this kind of thing,

> !!! [alpha, OK]] for "problem: lower bound alpha level not reached"

could actually work OK if done well.  The part of Tim's suggestion with which I'm least comfortable is reporting the actual values of dubious confidence limits instead of `NA`.  This happens to be a reversal of opinion.  Back when the current style of CI reporting in `summary()` was first implemented, I suggested instead reporting the obtained values of dodgy confidence limits and leaving it to the user to run `summary()`with `verbose=T` to find out the details.  Tim replied that no one would ever bother doing that, and I conceded he had a point.

**Comment by tbates** [comment]
To bump this idea of showing the user the CIs we find (instead of NA), while still showing them the !!! warning note.

Here's a simple self-contained example. You'll likely need to install from github, as this is umx 3.0 code.

`install_github("tbates/umx")`

```R
# 1. Simulate data and run ACE model
tmp = umx_make_TwinData(nMZpairs = 10000, AA = .30, CC = .00)
m1 = umxACE(selDVs = "var", data = tmp)
umxConfint(m1, run=TRUE)
```
The ACE model has lbounds on the diag. Two CIs are NA

```R
confidence intervals:
                 lbound     estimate    ubound note
top.a_std[1,1] 0.5123506 5.381624e-01        NA  !!!
top.c_std[1,1]        NA 2.042777e-05 0.1421773  !!!
top.e_std[1,1] 0.8332407 8.428411e-01 0.8524399     
```

Would have made sense to print the NA items:

4. CIdetail
```R
summary(m1, verbose=TRUE)$CIdetail
CI details:
      parameter  side     value      fit              diagnostic                       statusCode            method
1 top.a_std[1,1] lower 0.5123506 112563.1                 success                               OK neale-miller-1997
2 top.a_std[1,1] upper 0.5529119 112563.1   active box constraint                               OK neale-miller-1997

3 top.c_std[1,1] lower 0.0000000 112742.6 alpha level not reached infeasible non-linear constraint neale-miller-1997
4 top.c_std[1,1] upper 0.1421773 112563.1                 success infeasible non-linear constraint neale-miller-1997

5 top.e_std[1,1] lower 0.8332407 112563.1                 success infeasible non-linear constraint neale-miller-1997
6 top.e_std[1,1] upper 0.8524399 112563.1                 success infeasible non-linear constraint neale-miller-1997
```

Here's a user example, where they are getting NA for their CIs.

```R
confidence intervals:
                  lbound     estimate ubound note
top.a_std[1,1]        NA 5.374627e-06     NA  !!!
top.c_std[1,1]        NA 3.113769e-06     NA  !!!
top.e_std[1,1] 0.9302154 1.000000e+00     NA  !!!

```

But the values accessible from verbose are just fine.

```R
CI details:
       parameter  side        value      fit              diagnostic                       statusCode            method
1 top.a_std[1,1] lower 0.000000e+00 2456.039 alpha level not reached infeasible non-linear constraint neale-miller-1997
2 top.a_std[1,1] upper 3.670132e-01 2457.993   active box constraint infeasible non-linear constraint neale-miller-1997
3 top.c_std[1,1] lower 4.822349e-17 2454.540 alpha level not reached infeasible non-linear constraint neale-miller-1997
4 top.c_std[1,1] upper 2.658007e-01 2457.993   active box constraint infeasible non-linear constraint neale-miller-1997
5 top.e_std[1,1] lower 9.302154e-01 2457.993                 success infeasible non-linear constraint neale-miller-1997
6 top.e_std[1,1] upper 1.000000e+00 2456.735 alpha level not reached infeasible non-linear constraint neale-miller-1997
```

**Comment by tbates** [comment]
Receiving output with NA is a showstopper for users, so would be great to implement printing parameter estimates instead of replacing them with NA.

Here (and in #75) I think we had unanimity about:

1.  Stopping replacing parameter estimates with NA in summary.
2. Adding a simple numeric code alongside questionable rows like `!1` and a note at the bottom explaining this or just pointing people to `mxSummary(model, verbose=T)`

But just printing the parameter value instead of NA and leave the !!! warning as is would be a big step forward for users.

**Comment by mcneale** [comment]
I agree, let's try to stop the NAs for confidence intervals.  If it's at or near a bound, and if the e.g., 3.84 hasn't been reached, IMO it is still fine to report it, but with a !1 note to say that, 3.84 not acquired.  In reality, if the bound is hard then there is no way could the confidence interval be anywhere further out.

**Comment by RMKirkpatrick** [comment]
Should we discuss this during Friday's meeting?

**Comment by tbates** [comment]
@RMKirkpatrick That would be good.

**Comment by RMKirkpatrick** [comment]
This issue, plus issue #322 , came up during today's meeting.
